File: Algorithm/Leetcode/[M]128. Longest Consecutive Sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0)
            return 0;
        
        Arrays.sort(nums);
        
        int longest = 0;
        int curr = 1;
        
        for(int i = 1; i < nums.length; i++){
            if (nums[i] != nums[i-1]){
                if (nums[i] == nums[i-1] + 1) {
                    curr++;
                } else {
                    longest = Math.max(curr, longest);
                    curr = 1;
                }
            }
        }
        return Math.max(longest, curr);
    }
}

File: Algorithm/Leetcode/[M]128. Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums.sort()
        
        
        long = 0
        curr = 1
        #print(nums)
        for i in range(len(nums)):
            if nums[i] == nums[i-1] + 1:
                #print("IN")
                curr += 1
            elif nums[i] == nums[i-1]:
                pass
            else:
                curr = 1
            
            long = max(long, curr)

        
        return long
       # print(long)